cmd/remote-storage/app/server_test.go

```diff
@@ -365,6 +377,7 @@ func TestServerGRPCTLS(t *testing.T) {
 			server, err := NewServer(
 				serverOptions,
 				f,
+				f,
```

Comment:
a little awkward to have to pass the same factory in twice - should I break it out into two fake factories?